data/src/main/java/com/microsoft/azure/kusto/data/AadAuthenticationHelper.java

```diff
@@ -62,23 +65,29 @@ public AadAuthenticationHelper(@NotNull ConnectionStringBuilder csb) throws URIS
     }
 
     String acquireAccessToken() throws DataServiceException  {
-        try {
-            switch (authenticationType) {
-                case AAD_APPLICATION_KEY:
-                    return acquireAadApplicationAccessToken().getAccessToken();
-                case AAD_USERNAME_PASSWORD:
-                    return acquireAadUserAccessToken().getAccessToken();
-                case AAD_DEVICE_LOGIN:
-                    return acquireAccessTokenUsingDeviceCodeFlow().getAccessToken();
-                case AAD_APPLICATION_CERTIFICATE:
-                    return acquireWithClientCertificate().getAccessToken();
-                default:
-                    throw new DataServiceException("Authentication type: " + authenticationType.name() + " is invalid");
+        if(lastAuthenticationResult == null || lastAuthenticationResult.getExpiresOnDate().before(new Date(System.currentTimeMillis() + ONE_MINUTE_IN_MILLIS))) {
```

Comment:
What we actually need to do in case of existing and nearly expired tokens, is acquire a new token, using [acquireTokenByRefreshToken](https://github.com/Microsoft/azure-activedirectory-interactive-auth-library-for-java/blob/36aed916659ae176245129ebb10d4a87f6966d37/src/com/microsoftopentechnologies/auth/AuthenticationContext.java#L200)